libraries/botbuilder-ai-orchestrator/src/orchestratorRecognizer.ts

```diff
@@ -337,100 +345,137 @@ export class OrchestratorRecognizer extends AdaptiveRecognizer implements Orches
-                    throw new Error(`Model load failed - model folder ${fullModelFolder}.`);
-                }
-            }
-            OrchestratorRecognizer.orchestrator = orchestrator;
-        }
+        // Load snapshot and create resolver
+        this._resolver = this._orchestrator?.orchestrator.createLabelResolver(snapshot);
+    }
 
+    private async tryScoreEntitiesAsync(text: string, recognizerResult: RecognizerResult) {
+        // It's impossible to extract entities without a _resolver object.
         if (!this._resolver) {
-            const fullSnapshotPath = resolve(this._snapshotFile);
-            if (!existsSync(fullSnapshotPath)) {
-                throw new Error(`Snapshot file does not exist at ${fullSnapshotPath}.`);
-            }
-            // Load the snapshot
-            const snapshot: Uint8Array = readFileSync(fullSnapshotPath);
-
-            // Load snapshot and create resolver
-            this._resolver = OrchestratorRecognizer.orchestrator.createLabelResolver(snapshot);
+            return;
         }
-    }
 
-    private async tryScoreEntities(text: string, recognizerResult: RecognizerResult) {
+        // Entity extraction can be controlled by the ScoreEntities flag.
+        // NOTE: SHOULD consider removing this flag in the next major SDK release (V5).
         if (!this.scoreEntities) {
             return;
         }
 
-        const results = await this._resolver.score(text, LabelType.Entity);
-        if (!results) {
-            throw new Error(`Failed scoring entities for: ${text}`);
+        // The following check is necessary to ensure that the _resolver object
+        // is capable of entity exttraction. However, this check does not apply to
+        // an external, mock-up _resolver.
+        if (!this._isResolverExternal) {
+            if (!this._orchestrator || !this._orchestrator.isEntityExtractionCapable) {
+                return;
+            }
         }
 
-        // Add full entity recognition result as a 'entityResult' property
-        recognizerResult[this.entityProperty] = results;
-        if (results.length) {
-            recognizerResult.entities ??= {};
-
-            results.forEach((result: Result) => {
-                const entityType = result.label.name;
-
-                // add value
-                const values = recognizerResult.entities[entityType] ?? [];
-                recognizerResult.entities[entityType] = values;
-
-                const span = result.label.span;
-                const entityText = text.substr(span.offset, span.length);
-                values.push({
-                    type: entityType,
-                    score: result.score,
-                    text: entityText,
-                    start: span.offset,
-                    end: span.offset + span.length,
-                });
+        // As this method is TryScoreEntities, so it's best effort only, there should
+        // not be any exception thrown out of this method.
```

Comment:
Actually it was named TryScoreEntities because there might not be entities extracted. :-)

Reply:
Agreed, I still keep the original code (see below) of throwing an exception, but wrap it in an try-catch -- for the best effort and no entity scenario. 

            const results = await this._resolver.score(text, LabelType.Entity);
            if (!results) {
                throw new Error(`Failed scoring entities for: ${text}`);
            }